core/src/snapshot_packager_service.rs

```diff
@@ -87,10 +89,30 @@ impl SnapshotPackagerService {
                         // With exit backpressure, we will delay flushing snapshot storages
                         // until we receive a graceful exit request.
                         // Save the snapshot storages here, so we can flush later (as needed).
-                        teardown_state = Some(TeardownState {
-                            snapshot_slot: snapshot_package.slot,
-                            snapshot_storages: snapshot_package.snapshot_storages.clone(),
-                        });
+                        // For fastboot snapshot packages, the bank snapshot is saved and
+                        // the rest of the snapshotting process is skipped
+                        if snapshot_kind == SnapshotKind::Fastboot {
+                            teardown_state = Some(TeardownState {
+                                snapshot_slot: snapshot_package.slot,
+                                snapshot_storages: snapshot_package.snapshot_storages.clone(),
+                                bank_snapshot_package: Some(snapshot_package.bank_snapshot_package),
+                            });
+
+                            let handling_time = measure_handling.end_as_us();
+                            datapoint_info!(
+                                "snapshot_packager_service_fastboot",
+                                ("enqueued_time_us", enqueued_time.as_micros(), i64),
+                                ("handling_time_us", handling_time, i64),
+                            );
+
+                            continue;
+                        } else {
+                            teardown_state = Some(TeardownState {
+                                snapshot_slot: snapshot_package.slot,
+                                snapshot_storages: snapshot_package.snapshot_storages.clone(),
+                                bank_snapshot_package: None,
+                            });
```

Comment:
If there's no exit backpressure, then we should not do anything during teardown, as there's no guarantee that teardown will complete before the process is killed.

We only want to do teardown on graceful exit.